SparkPost/2web2ui
# Sa 1275 - Metrics date allows selecting precision

### What Changed
 - Everything
 - Added precision selector to the report options.
 - Added selectedPrecision state to Datepicker.
 - Changed logic in Manual Entry Form.
 - Changed many metrics helper functions to include precision as a variable.
 - Added new function in metrics helper.
 - Added new configs for recommended time and precision for metrics rollup

### How To Test
- Add `x-msys-metrics-rollup` as user option by making a PUT request to '/api/v1/users/{username}/control ' endpoint and with 
```json
{ "options" : { "ui" :{ "use-metrics-rollup": true}	}}
 ```  
 - Add X-Msys-Metrics-Rollup to allowed header in your cors file in openresty
 - Navigate to summary report.
 - Change data ranges and precision around. It should meet AC.
 - Go back to an account without feature. Verify that old functionality has not changed.
### To Do
- [ ] UX feedback
- [ ] Fix changes
